src/Workspaces/Core/Portable/Workspace/Solution/ProjectState_Checksum.cs

```diff
@@ -69,6 +69,7 @@ private async Task<ProjectStateChecksums> ComputeChecksumsAsync(CancellationToke
                     var analyzerConfigDocumentChecksums = new ChecksumCollection(await analyzerConfigDocumentChecksumTasks.WhenAll().ConfigureAwait(false));
 
                     return new ProjectStateChecksums(
+                        this.Id,
```

Comment:
added projectid to the projectstatechecksums.  this avoids a lot of roundtrips to jsut go from the checksum object to get the info checksum to get the info to get the id.